### ISSUE TITLE
STDDEV function not proprely working

### ISSUE THREAD

**Issue by profeta64** [MajenkoLibraries/Average#7]
Hello,
I'm using this library with an Arduino Mega.
I defined the object so:

    Average<int> avgintbuffer(intbuffer_size);

and then i push 4 values:

    avgintbuffer.push(5);
    avgintbuffer.push(5);
    avgintbuffer.push(5);
    avgintbuffer.push(5);
    Serial.println(avgintbuffer.mean());
    Serial.println(avgintbuffer.stddev());

I get as ouptut: 5 (mean), 2.5 (stddev)

The standard deviation should be 0....

**Comment by majenkotech** [comment]
TBH I have been suspicious of this function for a while now.  I forget where I got the algorithm, but I am suspecting it may be completely wrong.  I just haven't had time to delve into it to find out what's wrong yet. If a better algorithm can be found that I can easily drop in that would be great...

**Comment by profeta64** [comment]
I noticed also now that the minimum function deliver every time 0...

**Comment by Bort-Plate** [comment]
I just tried the code that profeta64 posted at the top, except I'm not sure what is going on with that first line:
`Average<int> avgintbuffer(intbuffer_size);`

I used:
`Average<int> ave(5);`

If I push five 5's, stddev is 0.
If I push four 5's, stddev is  2.5

I tried other combinations of push values and buffer sizes. As long as you have filled the buffer, stddev should be correct.

The code seems correct in regard to the formula for standard deviation.  
What I don't understand is *why* stddev is wrong when the buffer hasn't been filled, because it seems to be based on `_count`, just like mean(), except mean() doesn't seem to have this problem.

```
template <class T> float Average<T>::stddev() {
	float square;
	float sum;
	float mu;
	float theta;
	int i;

    if (_count == 0) {
        return 0;
    }

	mu = mean();

	sum = 0;
	for(uint32_t i = 0; i < _count; i++) {
		theta = mu - (float)get(i);
		square = theta * theta;
		sum += square;
	}
	return sqrt(sum/(float)_count);
}
```

**Comment by Bort-Plate** [comment]
Also, @profeta64, when you said "I noticed also now that the minimum function deliver every time 0...", did you mean the the minimum() function returns 0, or that stddev() returns a minimum of 0?   
Standard deviation should never be less than 0. If you issue is with the minimum function, perhaps that's best suited as a different "issue" on git.

**Comment by profeta64** [comment]
Oh, quite long time ago..
If i remember correctly, I was meaning that the minimum() function has this defect.
By pushing the same four times 5, you get as minimum 0.
This happens perhaps, because, as you mentioned, i was not filling the buffer...

I did not checked it.

**Comment by majenkotech** [comment]
I think this is most likely related to the get bug now fixed in 32a4f32

**Comment by Bort-Plate** [comment]
It appears to me that this STDDEV issue has indeed been fixed.
